nextcord/application_command.py

```diff
@@ -647,16 +811,64 @@ async def call(self, state: ConnectionState, interaction: Interaction, option_da
+        except Exception as error:
+            state.dispatch('application_error', interaction, error)
+            await app_cmd.invoke_error(interaction, error)
+            return
+
+        if can_run:
+            if app_cmd._application_before_invoke is not None:
+                await app_cmd._application_before_invoke(interaction)
+
+            cog_application_before_invoke = app_cmd.cog_application_before_invoke
+            if cog_application_before_invoke is not None:
+                await cog_application_before_invoke(interaction)
+
+            if state._application_before_invoke is not None:
+                await state._application_before_invoke(interaction)
+
+            invoke_error = None
+            try:
+                await callback(*args)
+            except Exception as error:
+                invoke_error = ApplicationInvokeError(error)
+
+            after_invoke_error = None
+            try:
+                if app_cmd._application_after_invoke is not None:
+                    await app_cmd._application_after_invoke(interaction)
+                
+                cog_application_after_invoke = app_cmd.cog_application_after_invoke
+                if cog_application_after_invoke is not None:
+                    await cog_application_after_invoke(interaction)
+
+                if state._application_after_invoke is not None:
+                    await state._application_after_invoke(interaction)
+            except Exception as error:
+                after_invoke_error = error
+
+            if invoke_error is not None:
+                state.dispatch('application_error', interaction, invoke_error)
+                await app_cmd.invoke_error(interaction, invoke_error)
+            if after_invoke_error is not None:
+                raise after_invoke_error
+
+    def _find_subcommand(self, option_data: List[Dict[str, Any]]) -> Tuple[Union[ApplicationSubcommand, ApplicationCommand], List[Dict[str, Any]]]:
         if self.children:
-            # Discord currently does not allow commands that have subcommands to be run. Therefore, if a command has
-            # children, a subcommand must be being called.
-            await self.children[option_data[0]["name"]].call(state, interaction, option_data[0].get("options", {}))
+            return self.children[option_data[0]["name"]]._find_subcommand(option_data[0].get("options", {}))
         elif self.type in (ApplicationCommandType.chat_input, ApplicationCommandOptionType.sub_command):
-            # Slash commands are able to have subcommands, therefore that is handled here.
-            await self.call_invoke_slash(state, interaction, option_data)
+            return self, option_data
         else:
-            # Anything that can't be handled in here should be raised for ApplicationCommand to handle.
-            # TODO: Figure out how to hide this in exception trace log, people don't need to see it.
```

Comment:
Why do you keep removing things which are there for a good reason? These comments don't seem to have been addressed, so why are they being removed?